apps/src/templates/NavigationBar.jsx

```diff
@@ -1,42 +1,100 @@
 import React, { PropTypes } from 'react';
+import _ from 'lodash';
 import color from "@cdo/apps/util/color";
+import FontAwesome from './FontAwesome';
 import NavigationBarLink from './NavigationBarLink';
 
 const styles = {
   navBar: {
     height: 50,
     width: '100%',
     backgroundColor: color.purple,
-    marginBottom: 20
+    marginBottom: 20,
+  },
+  linksBox: {
+    overflow: 'auto',
+    whiteSpace: 'nowrap',
+    display: 'inline',
+    float: 'left',
+    width: 920
   },
   linkBox: {
     display: 'inline',
     paddingRight: 20,
     paddingLeft: 20,
   },
+  chevron: {
+    fontSize: 20,
+    lineHeight: '50px',
+    color: color.white,
+    textDecoration: 'none',
+    cursor: 'pointer',
+    height: 50,
+    width: 25,
+    backgroundColor: color.purple,
+    paddingLeft: 10,
+    paddingRight: 10,
+    display: 'inline',
+    float: 'left'
+  }
 };
 
 export default class NavigationBar extends React.Component {
   static propTypes = {
     links: PropTypes.array,
   };
 
+  state = {
+    listPosition: "start"
+  };
+
+  scrollForward = () => {
+    const lastLink = _.last(this.props.links);
+    const element = document.getElementById(lastLink.url);
+    element.scrollIntoView({block: "end"});
+    this.setState({listPosition: "end"});
+  };
+
+  scrollBackward = () => {
+    const firstLink = _.first(this.props.links);
+    const element = document.getElementById(firstLink.url);
```

Comment:
I think I could. I might have to wrestle with it a bit since the `NavigationBar` would be looking for a ref in a child `NavigationBarLink`; I'd have to research more about how React handles the context of the ref and children, but should be doable.  Preferable because it's more React-y?  Are there other benefits?

Reply:
I think it's just the right way to do things in React.  Otherwise we're relying on React for some rendering but then making assumptions about what it has rendered when we go straight to the DOM.

Reply:
Reading about it more, refs do seem like the best choice here.  I'm going to merge now so that the Poorva/Mark can play with the scrolling and we can see how it handles really long translations.  I'll tackle the refactor as follow-up.  It's on my list, so hold me to it 😁